scripts/functest.sh

```diff
@@ -0,0 +1,122 @@
+#! /usr/bin/env bash
+# -*- coding: utf-8 -*-
+
+CSCLI_BIN="./cscli"
+CSCLI="${CSCLI_BIN} -c dev.yaml"
+JQ="jq -e"
+HTTP_URI="http://localhost:8081"
+
+
+function fail
+{
+    echo "test failure !"
+    caller 0
+    exit 1
+}
+
+function init
+{
+    killall crowdsec
+    rm data/crowdsec.db
+    ${CSCLI} machines add -a
+    ./crowdsec -c dev.yaml> crowdsec-out.log 2>&1  &
+    sleep 1
+
+}
+
+
+function docurl
+{
+    APIK=$1
+    URI=$2
+    curl -s -H "X-Api-Key: ${APIK}" "http://localhost:8081${URI}"
+} 
+
+###
+### ipv4 tests
+###
+
+init
+APIK=`${CSCLI} bouncers add TestingBouncer -o=raw`
+
+#
+# ip ipv4 tests
+#
+${CSCLI} decisions list -o json | ${JQ} '. == null' || fail
+docurl ${APIK} /v1/decisions | ${JQ} '. == null' || fail
+#add ip decision
+${CSCLI} decisions add -i 1.2.3.4  || fail
+${CSCLI} decisions list -o json | ${JQ} '.[].decisions[0].value == "1.2.3.4"'  || fail
+docurl ${APIK} /v1/decisions | ${JQ} '.[0].value == "1.2.3.4"' || fail
+#check ip match
+${CSCLI} decisions list -i 1.2.3.4 -o json | ${JQ} '.[].decisions[0].value == "1.2.3.4"'  || fail
+docurl ${APIK} /v1/decisions?ip=1.2.3.4 | ${JQ} '.[0].value == "1.2.3.4"' || fail
+${CSCLI} decisions list -i 1.2.3.5 -o json | ${JQ} '. == null'  || fail
+docurl ${APIK} /v1/decisions?ip=1.2.3.4 | ${JQ} '. == null' || fail
+#check outer range match
+${CSCLI} decisions list -r 1.2.3.0/24 -o json | ${JQ} '. == null'  || fail
+docurl ${APIK} "/v1/decisions?range=1.2.3.0/24" | ${JQ} '. == null' || fail
+${CSCLI} decisions list -r 1.2.3.0/24 --contained -o json |${JQ} '.[].decisions[0].value == "1.2.3.4"'  || fail
+docurl ${APIK} "/v1/decisions?range=1.2.3.0/24&contains=false" | ${JQ} '.[0].value == "1.2.3.4"' || fail
+#
+# range ipv4 tests
+#
+#add range decision
+${CSCLI} decisions add -r 4.4.4.0/24 || fail
+${CSCLI} decisions list -o json | ${JQ} '.[0].decisions[0].value == "4.4.4.0/24", .[1].decisions[0].value == "1.2.3.4"' || fail
+docurl ${APIK} "/v1/decisions" | ${JQ} '.[0].value == "4.4.4.0/24", .[1].value == "1.2.3.4"' || fail
```

Comment:
is order always maintained for '.[0].value == "4.4.4.0/24", .[1].value == "1.2.3.4"' ?

Reply:
It seems that it is maintained